tokio/src/time/driver/entry.rs

```diff
@@ -494,6 +494,10 @@ impl TimerEntry {
         unsafe { &*self.inner.get() }
     }
 
+    fn driver_shutdown(&self) -> bool {
+        self.driver.lock().is_shutdown
```

Comment:
Do we want to move this to some sort of `AtomicBool`?

Reply:
Yeah I was wondering if this lock could be bad. The `is_shutdown` is defined [here](https://github.com/tokio-rs/tokio/blob/master/tokio/src/time/driver/mod.rs#L149) which is accessible through an [`Arc<Mutex<_>>` here](https://github.com/tokio-rs/tokio/blob/master/tokio/src/time/driver/handle.rs#L9), therefore the lock.

I guess we could move [`Inner.is_shutdown`](https://github.com/tokio-rs/tokio/blob/master/tokio/src/time/driver/mod.rs#L135) to [`Handle`](https://github.com/tokio-rs/tokio/blob/master/tokio/src/time/driver/handle.rs#L7) and make it an `AtomicBool`. Is that what you have in mind?